packages/paste-core/components/sidebar/src/SidebarHeader.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from '@twilio-paste/box';
+import {Button} from '@twilio-paste/button';
+import type {BoxProps} from '@twilio-paste/box';
+import {ProductHomeIcon} from '@twilio-paste/icons/esm/ProductHomeIcon';
+
+import {SidebarContext} from './SidebarContext';
+
+export interface SidebarHeaderProps extends React.HTMLAttributes<HTMLButtonElement> {
+  productIcon?: React.ReactNode;
+  onClick?: () => void;
+  element?: BoxProps['element'];
+}
+
+export const SidebarHeader = React.forwardRef<HTMLButtonElement, SidebarHeaderProps>(
+  ({element = 'SIDEBAR_HEADER', productIcon, onClick, children}, ref) => {
+    const {collapsed} = React.useContext(SidebarContext);
+
+    return (
+      <Box
+        padding="space60"
+        borderBottomWidth="borderWidth10"
+        borderBottomStyle="solid"
+        borderBottomColor="colorBorderInverseWeaker"
+        display="flex"
+        alignItems="center"
+        justifyContent="flex-start"
+        columnGap="space50"
+        overflow="hidden"
+        element={element}
+        ref={ref}
+      >
+        <Button size="icon" variant="inverse" element={`${element}_BUTTON`} onClick={onClick}>
+          {productIcon ? productIcon : <ProductHomeIcon size="sizeIcon20" decorative element={`${element}_ICON`} />}
```

Comment:
suggestion: specify icon color

Reply:
I think `currentColor` as default should work fine. Any reason why you recommend specifying it?

Reply:
Does that allow a product to change the icon color? Like, say in the future if we do want each product to use the brand colors for the icon color?

Reply:
Yup. Teams can pass in their own icon entirely with the `productIcon` prop.